flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/testutil/HBaseTestCluster.java

```diff
@@ -0,0 +1,328 @@
+            return false;
+        }
+    }
+
+    public void clearTables() {
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            for (TableDescriptor table : admin.listTableDescriptors()) {
+                admin.disableTable(table.getTableName());
+                admin.deleteTable(table.getTableName());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Could not clear test cluster tables", e);
+        }
+    }
+
+    public void clearReplicationPeers() {
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            StringBuilder logMessage = new StringBuilder("Clearing existing replication peers:");
+            for (ReplicationPeerDescription desc : admin.listReplicationPeers()) {
+                logMessage.append("\n\t").append(desc.getPeerId()).append(" | ").append(desc);
+                admin.removeReplicationPeer(desc.getPeerId());
+            }
+            LOG.info(logMessage.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Could not clear test cluster replication peers", e);
+        }
+    }
+
+    public List<ReplicationPeerDescription> getReplicationPeers() {
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            return admin.listReplicationPeers();
+        } catch (IOException e) {
+            LOG.error("Error retrieving replication peers", e);
+            return null;
+        }
+    }
+
+    public void makeTable(String tableName) {
+        makeTable(tableName, 1);
+    }
+
+    /**
+     * Creates a table for given name with given number of column families. Column family names
+     * start with {@link HBaseTestCluster#COLUMN_FAMILY_BASE} and are indexed, if more than one is
+     * requested
+     */
+    public void makeTable(String tableName, int numColumnFamilies) {
+        assert numColumnFamilies >= 1;
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            TableName tableNameObj = TableName.valueOf(tableName);
+            if (!admin.tableExists(tableNameObj)) {
+                TableDescriptorBuilder tableBuilder =
+                        TableDescriptorBuilder.newBuilder(tableNameObj);
+                for (int i = 0; i < numColumnFamilies; i++) {
+                    ColumnFamilyDescriptorBuilder cfBuilder =
+                            ColumnFamilyDescriptorBuilder.newBuilder(
+                                    Bytes.toBytes(COLUMN_FAMILY_BASE + i));
+                    cfBuilder.setScope(1);
+                    tableBuilder.setColumnFamily(cfBuilder.build());
+                }
+                admin.createTable(tableBuilder.build());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Could not create test cluster table", e);
```

Comment:
Maybe it would be easier to just throw the IOException without handling? Alternatively failing the test with an AssertionError.

Reply:
✅ in b0e0f5d7a0bc332b81aa949d357de063f7a00d98